pandas/io/sql.py

```diff
@@ -139,24 +140,46 @@ def _parse_date_columns(data_frame, parse_dates):
     return data_frame
 
 
+def _convert_arrays_to_dataframe(
+    data,
+    columns,
+    coerce_float: bool = True,
+    use_nullable_dtypes: bool = False,
+) -> DataFrame:
+    content = lib.to_object_array_tuples(data)
+    arrays = convert_object_array(
+        list(content.T),
+        dtype=None,
+        coerce_float=coerce_float,
+        use_nullable_dtypes=use_nullable_dtypes,
+    )
```

Comment:
looks like this has gotten some Arrow-specific logic added here in the interim.  can we move the backend-specific stuff from core.internals here?